devtools/_git.py

```diff
@@ -0,0 +1,36 @@
+from typing import Generator
+from contextlib import contextmanager
+from subprocess import CompletedProcess
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from . import run
+
+class Git:
```

Comment:
Since `Git` is only used with its default argument we could consider limiting its use to that special case and turning its methods into module methods (similar to the logs) so we don't need to instantiate it - unless we anticipate this situation may change in future.

Reply:
`Git.worktree` makes a new instance of `Git`.

Reply:
Check